### DISCUSSION TITLE
Cause of "srtlib epoll.cpp:591:wait: : EID:1 no sockets to check, this would deadlock" messages?

### DISCUSSION THREAD

**Discussion by oripessach** [Haivision/srt#2969] [Q&A]
I'm trying to figure out a failure mode that results in a continuous stream of these messages:

"srtlib epoll.cpp:591:wait: : EID:1 no sockets to check, this would deadlock"

Looking at epoll.cpp it's clear that there are no sockets to check, as the message helpfully says, but It's not clear to me how we get into that state. Any insight on how the code works or where to look would be helpful.

Our code initiates a connection to a remote host in caller mode, and works for hours before this starts happening. In tests, I'm able to bring the remote host down and back up and the client simply reconnects.

In cases where these messages are output, the remote host stays up, and reports a timeout. What might be happening here?

**Comment by ethouris** [comment]
The only way how a socket can be withdrawn from the EID is by calling `srt_epoll_remove_usock` or when the socket is closed (by `srt_close`) or broken after being once successfully connected (not sockets that were used in `srt_connect` and failed to connect).

**Reply by oripessach** [reply]
OK. What do you mean by 'broken'? Since this is the situation I'm looking at (a stream is running fine for a while and then not working.)

I'm using gstreamer, by the way, and I'm basically debugging their srt support at this point... Any pointers would be helpful.

**Reply by RicardoTactiql** [reply]
Did you end up resolving this?
